specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,272 @@
+  ///
+  /// \snippet ViewComponent.cpp ToggleDefaultDownloadDialog
+  HRESULT OpenDefaultDownloadDialog();
+
+  /// Close the default download dialog.
+  HRESULT CloseDefaultDownloadDialog();
+
+  /// Get the default download dialog position.
+  [propget] HRESULT DefaultDownloadDialogPosition(
+      [out, retval] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION* position);
+
+  /// Set the default download dialog position relative to the WebView bounds.
+  /// The dialog can be positioned against any of the WebView corners (see
+  /// COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION). When the WebView or dialog
+  /// changes size, the dialog keeps its position relative to the corner. The
+  /// dialog may become partially or completely outside of the WebView bounds if
+  /// the WebView is small enough. Set the padding from the position with the
+  ///  `DefaultDownloadDialogPadding` property.
+  ///
+  /// \snippet ViewComponent.cpp SetDefaultDownloadDialogPosition
+  [propput] HRESULT DefaultDownloadDialogPosition(
+      [in] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION position);
+
+  /// Get the default download dialog padding.
+  [propget] HRESULT DefaultDownloadDialogPadding(
+      [out, retval] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_PADDING* padding);
+
+  /// Set the default download dialog padding from the
+  /// `DefaultDownloadDialogPosition`. See
+  /// COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_PADDING.
+  [propput] HRESULT DefaultDownloadDialogPadding(
+      [in] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_PADDING padding);
+}
+
+/// Implements the interface to receive `IsDefaultDownloadDialogOpenChanged`
+/// events.
+[uuid(86557854-794a-414a-b046-ba515f617306), object, pointer_default(unique)]
+interface ICoreWebView2IsDefaultDownloadDialogOpenChangedEventHandler :
+   IUnknown {
+  /// Provides the event args for the corresponding event. No event args exist
+  /// and the `args` parameter is set to `null`.
+  HRESULT Invoke([in] ICoreWebView2* sender, [in] IUnknown* args);
+}
+```
+
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2;
+
+    enum CoreWebView2DefaultDownloadDialogPosition
+    {
+        TopLeft = 0,
+        TopRight = 1,
+        BottomLeft = 2,
+        BottomRight = 3,
+    };
+
+    struct CoreWebView2DefaultDownloadDialogPadding
+    {
+        UInt32 x;
+        UInt32 y;
+    };
+
+    runtimeclass CoreWebView2
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_6")]
+        {
+            // ICoreWebView2_6 members
+            Boolean IsDefaultDownloadDialogOpen { get; };
+
+            event Windows.Foundation.TypedEventHandler<CoreWebView2, Object>
+                IsDefaultDownloadDialogOpenChanged;
+
+            void OpenDefaultDownloadDialog();
+
+            void CloseDefaultDownloadDialog();
+
+            CoreWebView2DefaultDownloadDialogPosition
+                DefaultDownloadDialogPosition { get; set; };
```

Comment:
`Position` sounds sort of misleading like its going to be a point. What about `Alignment` (or `CornerAlignment` or other ideas?)? In XAML there's HorizontalAlignment and VerticalAlignment properties that are enums that can have an element sort of fix itself to the different sides you set much like this case except they support many more options. I'm not sure if there's a corresponding precedent from HTML/CSS we can follow?

Reply:
Looks like CSS is similar to XAML: https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align
I think `CornerAlignment` makes sense in our case? Or do we want to match the precedent and use Horizontal/VerticalAlignment with less options?